src/controllers/lookup-table-handler.ts

```diff
@@ -0,0 +1,285 @@
+                    nonMatchingValues: nonMatchedValues.map((row) => Object.values(row)[0])
+                }
+            };
+        }
+        if (nonMatchedRows.length > 0) {
+            const nonMatchedValues = await quack.all(
+                `SELECT DISTINCT fact_table_column FROM (SELECT "${dimension.factTableColumn}" as fact_table_column FROM ${factTableName}) as fact_table LEFT JOIN ${lookupTableName} ON CAST(fact_table.fact_table_column AS VARCHAR)=CAST(${lookupTableName}."${confirmedJoinColumn}" AS VARCHAR) where lookup_table_column IS NULL;`
+            );
+            logger.error(
+                `The user supplied an incorrect or incomplete lookup table and ${nonMatchedRows.length} rows didn't match`
+            );
+            return {
+                status: 400,
+                dataset_id: dataset.id,
+                errors: [
+                    {
+                        field: 'patch',
+                        tag: { name: 'errors.dimensionValidation.invalid_lookup_table', params: {} },
+                        message: [
+                            {
+                                lang: Locale.English,
+                                message: t('errors.dimensionValidation.invalid_lookup_table', {
+                                    lng: Locale.English
+                                })
+                            }
+                        ]
+                    }
+                ],
+                extension: {
+                    totalNonMatching: nonMatchedRows.length,
+                    nonMatchingValues: nonMatchedValues.map((row) => Object.values(row)[0])
+                }
+            };
+        }
+    } catch (error) {
+        logger.error(
+            `Something went wrong, most likely an incorrect join column name, while trying to validate the lookup table with error: ${error}`
+        );
+        const nonMatchedRows = await quack.all(`SELECT COUNT(*) AS total_rows FROM ${factTableName};`);
+        const nonMatchedValues = await quack.all(`SELECT DISTINCT ${dimension.factTableColumn} FROM ${factTableName};`);
+        return {
+            status: 400,
+            dataset_id: dataset.id,
+            errors: [
+                {
+                    field: 'patch',
+                    tag: { name: 'errors.dimensionValidation.invalid_join_column', params: {} },
+                    message: [
+                        {
+                            lang: Locale.English,
+                            message: t('errors.dimensionValidation.invalid_join_column', {
+                                lng: Locale.English
+                            })
+                        }
+                    ]
+                }
+            ],
+            extension: {
+                totalNonMatching: nonMatchedRows[0].total_rows,
+                nonMatchingValues: nonMatchedValues.map((row) => Object.values(row)[0])
+            }
+        };
```

Comment:
So the frontend needs more than just an error in a good many cases... We can stop using the veiwErrDto but we'll just have to introduce another interface/class to send this information back to the router... Which then will turn this in to a viewErrDto for the frontend to consume as you need the error message/tag, the params and the potentially the preview or some other artefact to send back.

Also you want to know if it failed because of something the user did (400) or something that went generally wrong because of something we did (500).  I suspect you end up with a object like ViewErrDto just without the message component.

We can refactor this out and agree post MVP.  Lets get this pair of journeys over the line and when we have breathing room look at what we want to change.